commands/check_cloudformation.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+import os
+from pathlib import Path
+from subprocess import run
+
+import click
+
+from commands.bootstrap_cli import make_config
+from commands.exceptions.CheckCloudformationFailure import CheckCloudformationFailure
+from commands.copilot_cli import make_storage
+
+BASE_DIR = Path(__file__).parent.parent
+
+
+def valid_checks():
+    return {
+        "lint": lint,
+    }
+
+
+@click.group(invoke_without_command=True)
+@click.argument("checks", nargs=-1)
+@click.pass_context
+def check_cloudformation(ctx, checks):
+    """Runs the checks passed in the command arguments.\n
+    If no argument is passed, it will run all the checks.
+    """
+
+    if len(checks) == 0:
+        checks = valid_checks()
+        running_checks = "all"
+    else:
+        running_checks = ' & '.join(', '.join(checks).rsplit(', ', 1))
+
+    for check in checks:
+        if check not in valid_checks():
+            click.secho(f"""Invalid check requested "{check}" """, fg="red")
+            exit(1)
+
+    check_single_or_plural = f"""check{"s" if ("all" in checks or len(checks) > 1) else ""}"""
+
+
+    click.secho(f"""\n>>> Preparing CloudFormation templates\n""", fg="yellow")
+    os.chdir(f"{BASE_DIR}/tests/test-application")
+    ctx.invoke(make_config, config_file="bootstrap.yml")
+    ctx.invoke(make_storage, storage_config_file="storage.yml")
+
+    click.secho(f"""\n>>> Running {running_checks} {check_single_or_plural}\n""", fg="yellow")
+
+    failed_checks = []
+    for check_name in checks:
+        if (check_name in valid_checks().keys()):
+            try:
+                ctx.invoke(valid_checks()[check_name])
+            except CheckCloudformationFailure as error:
+                failed_checks.append(error)
+
+    if len(failed_checks) > 0:
+        click.secho("The CloudFormation templates did not pass the following checks:", fg="red")
+        for failed_check in failed_checks:
+            click.secho(f"  - {failed_check}", fg="red")
+        exit(1)
+    else:
+        click.secho("The CloudFormation templates passed all the checks :-)", fg="green")
+
+
+@check_cloudformation.command()
+def lint():
```

Comment:
I'm not sure how to unit test this one. Maybe we are happy with unit testing the main `check-cloudformation` command, which does all the orchestration?

Reply:
You could lint something and check that a string exists in the output?

Reply:
Maybe, but I'd have to get somehow put a known failing template in place. Currently all the addons are generated at the start of the command after completely removing the `tests/test-application/copilot` directory, so I think it might be a pain in the wotsit.